### PR TITLE
web: editable list view: resize columns

### PR DESCRIPTION
**Pull request by Arcasias**
Add a feature allowing columns resizing in editable lists

Task 2055903

### CODE REVIEW

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 1]
```diff
@@ -512,6 +527,23 @@ ListRenderer.include({
...
+     * There is no reverse method (list._removeEventListener) because there is
+     * no issue with removing an non-existing listener.
+     * 
+     * @private
+     * @param {EventTarget} el
+     * @param {string} name event name
+     * @param {Function} callback callback function to attach
+     * @param {Object} options event listener options
+     */
+    _addEventListener: function (type, el, callback, options) {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: docstring doesn't match the API

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 2]
```diff
@@ -556,12 +588,20 @@ ListRenderer.include({
...
-            var $th = $(th);
-            $th.css('width', restoreWidths ? this.columnWidths[index] : $th.outerWidth() + 'px');
+        // Freeze table width according to its size in fixed layout
+        const table = this.el.getElementsByTagName('table')[0];
+        const thead = table.getElementsByTagName('thead')[0];
+        table.style.tableLayout = 'fixed';
+        table.style.width = `${table.offsetWidth}px`;
+
+        // Freeze each th width according to their size in auto layout
+        table.style.tableLayout = 'auto';
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: isn't it already the case?

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 3]
```diff
@@ -966,8 +1006,28 @@ ListRenderer.include({
...
+     * Overriden to add a resize handle in editable list column headers.
+     * Only applies to data column headers
+     * 
+     * @override
+     * @private
+     */
+    _renderHeaderCell: function () {
+        const $th = this._super.apply(this, arguments);
+        const name = $th.data('name');
+        if (this.editable && this._hasVisibleRecords(this.state) && name && name !== 'sequence') {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: I don't know if we want to handle those special cases here (there are probably other widgets for which resizing is pointless). We may probably let the user do it if he wants (or is there a real issue with it?).

File: addons/web/static/src/scss/list_view.scss [hunk 4]
```diff
@@ -180,10 +184,39 @@
...
         &.o_editable_list {
+            &.o_resizing tr > :not(.o_column_resizing) {
+                opacity: 0.5;
+            }
+
+            th.o_column_resizing {
+                background-color: #dfdfdf;
+                span.o_resize {
+                    border-right: 3px dashed $o-brand-primary;
+                }
```
Comment (aab-odoo) [inline addons/web/static/src/scss/list_view.scss]: maybe not necessary (to check with a PO)

File: addons/web/static/src/scss/list_view.scss [hunk 5]
```diff
@@ -180,10 +184,39 @@
...
             &.o_empty_list {
                 table-layout: fixed;
             }
 
+            &:not(.o_empty_list):not(.o_resizing) {
+                th:hover {
+                    background-color: #dfdfdf;
+                    span.o_resize {
+                        border-right: 3px dashed $o-brand-primary;
+                    }
```
Comment (aab-odoo) [inline addons/web/static/src/scss/list_view.scss]: same

File: addons/web/static/src/js/views/list/list_editable_renderer.js (hunk 2 above)
Comment (Arcasias) [reply addons/web/static/src/js/views/list/list_editable_renderer.js]: The table in `fixed` layout adapt its width according to its parent, while it adapts it to its children whil in `auto`. To have the final width, we have to set it to fix, measure it, and then reset it to auto to measure the column widths.

**Comment by aab-odoo** [comment]
robodoo rebase-ff

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 6]
```diff
@@ -110,6 +113,18 @@ ListRenderer.include({
         }
         return this._super();
     },
+    /**
+     * Overriden to unbind all attached listeners
+     *
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: there is a trailing space here

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 7]
```diff
@@ -511,6 +526,23 @@ ListRenderer.include({
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * Used to bind event listeners so that they can be unbound when the list
+     * is destroyed.
+     * There is no reverse method (list._removeEventListener) because there is
+     * no issue with removing an non-existing listener.
+     *
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: trailing space

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 8]
```diff
@@ -511,6 +526,23 @@ ListRenderer.include({
...
+     * @private
+     * @param {string} type event name
+     * @param {EventTarget} el event target
+     * @param {Function} callback callback function to attach
+     * @param {Object} options event listener options
+     */
+    _addEventListener: function (type, el, callback, options) {
+        el.addEventListener(type, callback, options);
+        this.eventListeners.push({ type, el, callback, options });
+    },
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: trailing space

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 9]
```diff
@@ -1371,11 +1440,101 @@ ListRenderer.include({
      * @override
      * @private
      */
-    _onSortColumn: function () {
-        if (this.currentRow === null) {
+    _onSortColumn: function (ev) {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: ev not used

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo retry

**Comment by aab-odoo** [comment]
robodoo retry

**Comment by Bonainifo** [comment]
It seems not avaliable in saas12.5

**Comment by aab-odoo** [comment]
@Bonainifo It is only for editable lists

**Pull request closed without merging**